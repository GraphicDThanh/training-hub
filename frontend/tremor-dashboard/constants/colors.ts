type ColorChannelsChartType = {
  [key: number]: string;
  1: string;
  2: string;
  3: string;
  4: string;
};

export const COLOR_CHANNELS_CHART: ColorChannelsChartType = {
  1: "blue-500",
  2: "red-500",
  3: "primary",
  4: "gray-500",
};
