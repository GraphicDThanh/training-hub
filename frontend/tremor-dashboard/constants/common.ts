// Types
import { OptionType } from "@/types";

export const EXCEPT_KEYS = {
  POSITIVE_INTEGER: ["e", "E", "+", "-", "."],
  POSITIVE_DOUBLE: ["e", "E", "+", "-"],
};

export enum DIRECTION {
  ASC = "asc",
  DESC = "desc",
}

// Mailto Support
export const MAILTO_SUPPORT = "support@tremor-dashboard.com";

export const REMEMBER_ME_COOKIES_KEY = "remember-me";
export const UID_KEY = "uid";
export const ROLE = "role";
export const NAME = "name";
export const AVATAR = "avatar";
export const AUTH_SESSION_COOKIES_KEY = "authjs.session-token";

export const ORDER_LIST_OPTIONS: OptionType[] = [
  { option: "Paid", value: "0" },
  { option: "Refunded", value: "2" },
  { option: "Canceled", value: "1" },
];

export const PRODUCT_LIST_OPTIONS: OptionType[] = [
  { option: "Yes", value: "true" },
  { option: "No", value: "false" },
];

export const FORMAT_DATE = {
  YEAR_MONTH_DATE_TIME: "YYYY-MM-DDTHH:mm:ss.SSS",
  MONTH_DAY_YEAR_TIME: "MMM D, YYYY h:mm A",
};

export const SEPARATOR = {
  COMMAS: ",",
  DOTS: ".",
};

export const CDN_KEY = process.env.NEXT_PUBLIC_CDN_KEY;

export const DRAG_ZONE = {
  DEFAULT: {
    TEXT: "Drag and drop your file here to begin uploading.",
    STYLE_INPUT: "border-gray-300",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_DRAG: {
    TEXT: "Release the mouse click to upload the photo",
    STYLE_INPUT: "outline outline-2 outline-[#ebeff4]",
    STYLE_TEXT: "text-gray-500 dark:text-gray-400",
  },
  ON_WRONG_FORMAT: {
    TEXT: "Invalid image file type. Please try uploading again.",
    STYLE_INPUT: "border-red-500",
    STYLE_TEXT: "text-red-500 dark:text-red-500",
  },
};

export const TRANSACTION_CLASS = {
  INCREASE: {
    buttonClasses: "text-few dark:text-few dark:hove:text-few border-few",
    textClasses: "font-semibold text-greener dark:text-few",
  },
  DECREASE: {
    buttonClasses:
      "text-attention dark:text-attention dark:hover:text-attention border-attention",
    textClasses: "font-semibold text-blood dark:text-pink",
  },
  PENDING: {
    buttonClasses:
      "text-primary dark:text-primary dark:hover:text-primary border-primary",
    textClasses:
      "font-semibold text-primary dark:text-dark-tremor-content-pending",
  },
};

export const enum VARIANT_BUTTON {
  LIGHT = "light",
  LIGHT_CARD = "lightCard",
  LIGHT_CENTER = "lightCenter",
  LIGHT_STATUS = "lightStatus",
  SECONDARY = "secondary",
  SECONDARY_SHADOW = "secondaryShadow",
  PRIMARY = "primary",
  PRIMARY_CENTER = "primaryCenter",
  SURFACE = "surface",
  DARK = "dark",
  ACTIVE = "active",
}

export const NO_IMAGE = "No Image";

export const RESULT_NOT_FOUND = "Result Not Found";

export const AMOUNT_LIMIT = 20000;
export const PRICE_LIMIT = 20000;
export const QUANTITY_LIMIT = 10000;
export const WEIGHT_LIMIT = 2000;

export const PREFIX_PHONE_NUMBER = "+1";

export const STORAGES = {
  IS_DARK_THEME: "isDarkTheme",
};

export const THEMES = {
  LIGHT: "light",
  DARK: "dark",
};

export const LOGO_TITLE = "Tremor Dashboard PRO";
