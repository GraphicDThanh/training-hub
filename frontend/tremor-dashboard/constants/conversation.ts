export const CONVERSATION_TABLES = {
  GROUPS: "groups",
  CHATS: "chats",
};

export const CONVERSATION_ORDERS = {
  CREATED_AT: "createdAt",
  UPDATED_AT: "updatedAt",
};

export const CONVERSATION_QUERYS = {
  USER_IDS: "userIds",
};

export enum CONVERSATION_STATUS {
  INITIAL = "initial",
  FETCHED = "fetched",
}

export const ADMIN_ID = parseInt(process.env.NEXT_PUBLIC_ADMIN_ID!);

export enum DOCUMENT_CHANGE_TYPE {
  ADDED = "added",
  MODIFIED = "modified",
  REMOVED = "removed",
}

export enum WHERE_FILTER_OP {
  ARRAY_CONTAINS = "array-contains",
}
