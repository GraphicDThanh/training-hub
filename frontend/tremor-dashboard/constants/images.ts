export const NOT_FOUND_IMAGE = "/images/no-image-square.webp";
export const LOGO_IMAGE = "/assets/images/sidebar-logo.webp";
export const PLACEHOLDER_IMAGE = "/images/image-regular.svg";

export const AVATAR_IMAGE = {
  XS: "/images/avatar/avatar-xs.webp",
  SM: "/images/avatar/avatar-sm.webp",
  MD: "/images/avatar/avatar-md.webp",
  LG: "/images/avatar/avatar-lg.webp",
};

export const METADATA_IMAGE = {
  HOME_PREVIEW: "/assets/images/home-preview.webp",
  MASTERCARD: "/assets/images/mastercard.webp",
};

export const PROJECT_IMAGE = "/assets/images/profile-project.webp";
