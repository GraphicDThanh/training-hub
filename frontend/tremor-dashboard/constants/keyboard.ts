export const KEYBOARD_KEYS = {
  BACKSPACE: "Backspace",
  DIGIT_1: "1",
};
