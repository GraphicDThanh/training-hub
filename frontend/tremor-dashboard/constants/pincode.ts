export const PIN_CODE_DEFAULT = "*";
export const PIN_CODE_LENGTH = 6;
export const PIN_CODE_SKIP_SET_KEY = "skipSetupPinCode";
