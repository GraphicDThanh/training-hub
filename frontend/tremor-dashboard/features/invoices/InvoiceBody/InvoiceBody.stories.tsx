import type { Meta, StoryObj } from "@storybook/react";

// Components
import InvoiceBody from "./InvoiceBody";
import { Card } from "@tremor/react";

// Constants
import { MOCK_INVOICE_DATA } from "@/mocks";

const meta = {
  title: "Components/Invoices/InvoiceBody",
  component: InvoiceBody,
  tags: ["autodocs"],
  argTypes: {
    id: { description: "Id of invoice detail body" },
    createdAt: { description: "Create date of invoice detail body" },
    dueAt: { description: "Due date of invoice detail body" },
    products: { description: "Products of invoice detail body" },
    totalCost: { description: "Total cost of invoice detail body" },
  },
} as Meta<typeof InvoiceBody>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  decorators: [
    Story => (
      <Card className="w-full dark:bg-dark-blue p-6 ring-0 rounded-xl shadow-md">
        <Story />
      </Card>
    ),
  ],
  args: {
    id: MOCK_INVOICE_DATA.id,
    createdAt: MOCK_INVOICE_DATA.createdAt,
    dueAt: MOCK_INVOICE_DATA.dueAt,
    products: MOCK_INVOICE_DATA.products,
    totalCost: MOCK_INVOICE_DATA.totalCost,
  },
};
