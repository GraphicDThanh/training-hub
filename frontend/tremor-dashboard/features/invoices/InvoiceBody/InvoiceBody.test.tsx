import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

// Components
import InvoiceBody from "./InvoiceBody";

// Mocks
import { MOCK_INVOICE_DATA } from "@/mocks";

// Constants
import { RESULT_NOT_FOUND } from "@/constants";
import { TInvoiceDetail } from "@/types";

jest.mock("next/navigation", () => ({
  usePathname: jest.fn().mockReturnValue("/mocked-path"),
  useRouter: jest.fn().mockReturnValue({
    replace: jest.fn(),
  }),
  useSearchParams: jest.fn().mockReturnValue(new URLSearchParams()),
}));

describe("InvoiceBody Testing", () => {
  const invoice = {
    id: MOCK_INVOICE_DATA.id,
    createdAt: MOCK_INVOICE_DATA.createdAt,
    dueAt: MOCK_INVOICE_DATA.dueAt,
    products: MOCK_INVOICE_DATA.products,
    totalCost: MOCK_INVOICE_DATA.totalCost,
  };
  it("should match snapshot", () => {
    const component = render(<InvoiceBody {...invoice} />);
    expect(component).toMatchSnapshot();
  });

  it("renders invoice body with default data show text Result Not Found", () => {
    const { getAllByText } = render(
      <InvoiceBody {...invoice} products={[] as TInvoiceDetail[]} />,
    );

    expect(getAllByText(RESULT_NOT_FOUND)).toHaveLength(1);
  });
});
