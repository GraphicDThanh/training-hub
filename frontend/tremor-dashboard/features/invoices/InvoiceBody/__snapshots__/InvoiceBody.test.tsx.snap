// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceBody Testing should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tremor-Flex-root flex w-full flex-col justify-between items-end invoice-body"
      >
        <div
          class="tremor-Flex-root flex w-full flex-col justify-between items-start mb-7 md:flex-row md:items-center print:flex-row print:mb-7"
        >
          <div
            class="w-full mb-5 md:mb-0 print:mb-0"
          >
            <p
              class="text-tremor-title font-normal dark:text-lighter text-tertiary print:text-greyer dark:print:text-greyer print:mb-1"
            >
              Invoice no
            </p>
            <a
              class="text-tremor-primary text-primary dark:text-white leading-relaxed font-bold tracking-wide no-underline print:text-primary dark:print:!text-primary"
              href="#"
            >
              #
              230220
            </a>
          </div>
          <div
            class="w-full"
          >
            <div
              class="tremor-Flex-root flex w-full flex-col justify-between items-center md:flex-row mb-1 md:mb-0 print:flex-row print:mb-1"
            >
              <p
                class="text-tremor-title w-full mb-1 md:mb-0 print:mb-1 text-left md:text-end print:text-end font-normal dark:text-lighter text-tertiary print:text-greyer dark:print:text-greyer"
              >
                Invoice date:
              </p>
              <p
                class="w-full text-left md:text-end print:text-end text-tremor-title text-primary font-semibold dark:text-white dark:print:text-primary"
              >
                03/06/2019
              </p>
            </div>
            <div
              class="tremor-Flex-root flex w-full flex-col justify-between items-center md:flex-row print:flex-row"
            >
              <p
                class="text-tremor-title w-full mb-1 md:mb-0 print:mb-1 text-left md:text-end print:text-end font-normal dark:text-lighter text-tertiary print:text-greyer dark:print:text-greyer"
              >
                Due date:
              </p>
              <p
                class="w-full text-left md:text-end print:text-end text-tremor-title text-primary font-semibold dark:text-white dark:print:text-primary"
              >
                03/11/2019
              </p>
            </div>
          </div>
        </div>
        <div
          class="relative w-full"
        >
          <div
            class="tremor-Card-root relative w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-0 border-none ring-0 dark:bg-dark-tremor-primary overflow-x-auto !shadow-none"
          >
            <div
              class="flex flex-col items-start justify-start my-2"
            >
              <div
                class="tremor-Table-root overflow-auto w-full"
              >
                <table
                  class="tremor-Table-table w-full text-tremor-default text-tremor-content dark:text-dark-tremor-content"
                  id="table"
                  tabindex="0"
                >
                  <thead
                    class="tremor-TableHead-root text-left text-tremor-content dark:text-dark-tremor-content"
                  >
                    <tr
                      class="tremor-TableRow-row"
                    >
                      <th
                        class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold uppercase cursor-default"
                        data-testid="header-cell-Item"
                      >
                        <div
                          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                        >
                          <div
                            class="tremor-Flex-root flex w-full flex-row justify-between items-center text-primary dark:text-white print:text-primary dark:print:!text-primary"
                          >
                            Item
                          </div>
                        </div>
                      </th>
                      <th
                        class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold uppercase cursor-default"
                        data-testid="header-cell-Qty"
                      >
                        <div
                          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                        >
                          <div
                            class="tremor-Flex-root flex w-full flex-row justify-between items-center text-primary dark:text-white print:text-primary dark:print:!text-primary"
                          >
                            Qty
                          </div>
                        </div>
                      </th>
                      <th
                        class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold uppercase cursor-default"
                        data-testid="header-cell-Rate"
                      >
                        <div
                          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                        >
                          <div
                            class="tremor-Flex-root flex w-full flex-row justify-between items-center text-primary dark:text-white print:text-primary dark:print:!text-primary"
                          >
                            Rate
                          </div>
                        </div>
                      </th>
                      <th
                        class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold uppercase cursor-default"
                        data-testid="header-cell-Amount"
                      >
                        <div
                          class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                        >
                          <div
                            class="tremor-Flex-root flex w-full flex-row justify-between items-center text-primary dark:text-white print:text-primary dark:print:!text-primary"
                          >
                            Amount
                          </div>
                        </div>
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="tremor-TableBody-root align-top divide-y divide-tremor-border dark:divide-dark-tremor-border"
                  >
                    <tr
                      class="tremor-TableRow-row"
                    >
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <p
                          class="text-tremor-default dark:text-dark-tremor-content max-w-[250px] md:max-w-sm truncate print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          Premium Support
                        </p>
                      </td>
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <p
                          class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          1
                        </p>
                      </td>
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <div>
                          <p
                            class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                          >
                            $ 9.00
                          </p>
                          <p
                            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 dark:!text-white dark:print:!text-primary"
                          >
                            Total
                          </p>
                        </div>
                      </td>
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <div>
                          <p
                            class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                          >
                            $ 9.00
                          </p>
                          <p
                            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 flex-col items-end dark:!text-white dark:print:!text-primary"
                          >
                            $345.00
                          </p>
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="tremor-TableRow-row"
                    >
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <p
                          class="text-tremor-default dark:text-dark-tremor-content max-w-[250px] md:max-w-sm truncate print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          Metro Dashboard
                        </p>
                      </td>
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <p
                          class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          3
                        </p>
                      </td>
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <div>
                          <p
                            class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                          >
                            $ 100.00
                          </p>
                          <p
                            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 dark:!text-white dark:print:!text-primary"
                          >
                            Total
                          </p>
                        </div>
                      </td>
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <div>
                          <p
                            class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                          >
                            $ 300.00
                          </p>
                          <p
                            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 flex-col items-end dark:!text-white dark:print:!text-primary"
                          >
                            $345.00
                          </p>
                        </div>
                      </td>
                    </tr>
                    <tr
                      class="tremor-TableRow-row"
                    >
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <p
                          class="text-tremor-default dark:text-dark-tremor-content max-w-[250px] md:max-w-sm truncate print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          Parts for service
                        </p>
                      </td>
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <p
                          class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          2
                        </p>
                      </td>
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <div>
                          <p
                            class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                          >
                            $ 10.00
                          </p>
                          <p
                            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 dark:!text-white dark:print:!text-primary"
                          >
                            Total
                          </p>
                        </div>
                      </td>
                      <td
                        class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                      >
                        <div>
                          <p
                            class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                          >
                            $ 20.00
                          </p>
                          <p
                            class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 flex-col items-end dark:!text-white dark:print:!text-primary"
                          >
                            $345.00
                          </p>
                        </div>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tremor-Flex-root flex w-full flex-col justify-between items-end invoice-body"
    >
      <div
        class="tremor-Flex-root flex w-full flex-col justify-between items-start mb-7 md:flex-row md:items-center print:flex-row print:mb-7"
      >
        <div
          class="w-full mb-5 md:mb-0 print:mb-0"
        >
          <p
            class="text-tremor-title font-normal dark:text-lighter text-tertiary print:text-greyer dark:print:text-greyer print:mb-1"
          >
            Invoice no
          </p>
          <a
            class="text-tremor-primary text-primary dark:text-white leading-relaxed font-bold tracking-wide no-underline print:text-primary dark:print:!text-primary"
            href="#"
          >
            #
            230220
          </a>
        </div>
        <div
          class="w-full"
        >
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-center md:flex-row mb-1 md:mb-0 print:flex-row print:mb-1"
          >
            <p
              class="text-tremor-title w-full mb-1 md:mb-0 print:mb-1 text-left md:text-end print:text-end font-normal dark:text-lighter text-tertiary print:text-greyer dark:print:text-greyer"
            >
              Invoice date:
            </p>
            <p
              class="w-full text-left md:text-end print:text-end text-tremor-title text-primary font-semibold dark:text-white dark:print:text-primary"
            >
              03/06/2019
            </p>
          </div>
          <div
            class="tremor-Flex-root flex w-full flex-col justify-between items-center md:flex-row print:flex-row"
          >
            <p
              class="text-tremor-title w-full mb-1 md:mb-0 print:mb-1 text-left md:text-end print:text-end font-normal dark:text-lighter text-tertiary print:text-greyer dark:print:text-greyer"
            >
              Due date:
            </p>
            <p
              class="w-full text-left md:text-end print:text-end text-tremor-title text-primary font-semibold dark:text-white dark:print:text-primary"
            >
              03/11/2019
            </p>
          </div>
        </div>
      </div>
      <div
        class="relative w-full"
      >
        <div
          class="tremor-Card-root relative w-full text-left rounded-tremor-default bg-tremor-background ring-tremor-ring shadow-tremor-card dark:ring-dark-tremor-ring dark:shadow-dark-tremor-card border-tremor-brand dark:border-dark-tremor-brand p-0 border-none ring-0 dark:bg-dark-tremor-primary overflow-x-auto !shadow-none"
        >
          <div
            class="flex flex-col items-start justify-start my-2"
          >
            <div
              class="tremor-Table-root overflow-auto w-full"
            >
              <table
                class="tremor-Table-table w-full text-tremor-default text-tremor-content dark:text-dark-tremor-content"
                id="table"
                tabindex="0"
              >
                <thead
                  class="tremor-TableHead-root text-left text-tremor-content dark:text-dark-tremor-content"
                >
                  <tr
                    class="tremor-TableRow-row"
                  >
                    <th
                      class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold uppercase cursor-default"
                      data-testid="header-cell-Item"
                    >
                      <div
                        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                      >
                        <div
                          class="tremor-Flex-root flex w-full flex-row justify-between items-center text-primary dark:text-white print:text-primary dark:print:!text-primary"
                        >
                          Item
                        </div>
                      </div>
                    </th>
                    <th
                      class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold uppercase cursor-default"
                      data-testid="header-cell-Qty"
                    >
                      <div
                        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                      >
                        <div
                          class="tremor-Flex-root flex w-full flex-row justify-between items-center text-primary dark:text-white print:text-primary dark:print:!text-primary"
                        >
                          Qty
                        </div>
                      </div>
                    </th>
                    <th
                      class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold uppercase cursor-default"
                      data-testid="header-cell-Rate"
                    >
                      <div
                        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                      >
                        <div
                          class="tremor-Flex-root flex w-full flex-row justify-between items-center text-primary dark:text-white print:text-primary dark:print:!text-primary"
                        >
                          Rate
                        </div>
                      </div>
                    </th>
                    <th
                      class="tremor-TableHeaderCell-root whitespace-nowrap text-left top-0 text-tremor-content-strong px-6 py-2 text-[10.4px] leading-[17px] dark:text-white tracking-[0.2px] font-bold uppercase cursor-default"
                      data-testid="header-cell-Amount"
                    >
                      <div
                        class="tremor-Flex-root flex w-full flex-row justify-between items-center"
                      >
                        <div
                          class="tremor-Flex-root flex w-full flex-row justify-between items-center text-primary dark:text-white print:text-primary dark:print:!text-primary"
                        >
                          Amount
                        </div>
                      </div>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="tremor-TableBody-root align-top divide-y divide-tremor-border dark:divide-dark-tremor-border"
                >
                  <tr
                    class="tremor-TableRow-row"
                  >
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <p
                        class="text-tremor-default dark:text-dark-tremor-content max-w-[250px] md:max-w-sm truncate print:text-greyer dark:print:text-greyer text-tertiary"
                      >
                        Premium Support
                      </p>
                    </td>
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <p
                        class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                      >
                        1
                      </p>
                    </td>
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <div>
                        <p
                          class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          $ 9.00
                        </p>
                        <p
                          class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 dark:!text-white dark:print:!text-primary"
                        >
                          Total
                        </p>
                      </div>
                    </td>
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <div>
                        <p
                          class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          $ 9.00
                        </p>
                        <p
                          class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 flex-col items-end dark:!text-white dark:print:!text-primary"
                        >
                          $345.00
                        </p>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="tremor-TableRow-row"
                  >
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <p
                        class="text-tremor-default dark:text-dark-tremor-content max-w-[250px] md:max-w-sm truncate print:text-greyer dark:print:text-greyer text-tertiary"
                      >
                        Metro Dashboard
                      </p>
                    </td>
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <p
                        class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                      >
                        3
                      </p>
                    </td>
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <div>
                        <p
                          class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          $ 100.00
                        </p>
                        <p
                          class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 dark:!text-white dark:print:!text-primary"
                        >
                          Total
                        </p>
                      </div>
                    </td>
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <div>
                        <p
                          class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          $ 300.00
                        </p>
                        <p
                          class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 flex-col items-end dark:!text-white dark:print:!text-primary"
                        >
                          $345.00
                        </p>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="tremor-TableRow-row"
                  >
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <p
                        class="text-tremor-default dark:text-dark-tremor-content max-w-[250px] md:max-w-sm truncate print:text-greyer dark:print:text-greyer text-tertiary"
                      >
                        Parts for service
                      </p>
                    </td>
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <p
                        class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                      >
                        2
                      </p>
                    </td>
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <div>
                        <p
                          class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          $ 10.00
                        </p>
                        <p
                          class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 dark:!text-white dark:print:!text-primary"
                        >
                          Total
                        </p>
                      </div>
                    </td>
                    <td
                      class="tremor-TableCell-root align-middle whitespace-nowrap text-left p-4 px-6 py-5 border-0 dark:border-grayish border-b border-gray-100"
                    >
                      <div>
                        <p
                          class="text-tremor-default dark:text-lighter print:text-greyer dark:print:text-greyer text-tertiary"
                        >
                          $ 20.00
                        </p>
                        <p
                          class="text-tremor-default text-tremor-content dark:text-dark-tremor-content total hidden mt-6 flex-col items-end dark:!text-white dark:print:!text-primary"
                        >
                          $345.00
                        </p>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
